### ISSUE TITLE
New rule: Ensure consistent LINQ style

### ISSUE THREAD

**Issue by Greg-Smulko** [meziantou/Meziantou.Analyzer#425]
**Is your feature request related to a problem? Please describe.**

I want to ensure that all LINQ queries across the solution are written in the same syntax - in my case method syntax rather than query syntax.

So, instead of having a mix of:
```cs
var meth = list.Where(x => x.Contains("sth"));

var quer = from l in list
           where l.Contains("sth_else")
           select l;
```

I'd like always to ensure that method syntax is used:
```cs
var meth = list.Where(x => x.Contains("sth"));
var quer = list.Where(x => x.Contains("sth_else"));
```

**Describe the solution you'd like**

I want all query-syntax LINQ usages to be flagged.

**Describe alternatives you've considered**

Do nothing, accept inconsistencies. 🤷 

**Additional context**

_none_

**Comment by meziantou** [comment]
I'm not sure this is something I want to implement in this project. I think it should be in Roslyn as this is where coding style rules are enforced (editorconfig + IDE1234 rules). Maybe you can try to open an issue on Roslyn and see if they want to implement it: https://github.com/dotnet/roslyn/issues/new/choose

**Comment by meziantou** [comment]
If Roslyn doesn't want to implement the feature and if someone want to implement it in this project, I will review and accept the PR as long as the rule is disabled by default (to avoid noise for something that is not really relevant for many projects).

**Comment by Greg-Smulko** [comment]
Thanks for the tip, here it is: https://github.com/dotnet/roslyn/issues/65788. Let's see...
